### DISCUSSION TITLE
Bug with Timer App + Vercel

### DISCUSSION THREAD

**Discussion by DennisWarchocki** [neuefische/questions#227] [Web] [answered]
Hello guys, 

Yesterday I finished my feature and everything was working fine, after I tryed some small changes I got a big Bug now because when the Seconds in the Timer going 1 second down, and then 1 Second up again. I have no Idea why this behave like that.

and Also When Im in my Preview in Vercel, I cant route to my timer page it says Error 404 to me if I type /timer after my Vercel url.

Vercel:
https://capstone-project-mgq8boic7-denniswarchocki.vercel.app   > /timer <
<img width="537" alt="Bildschirmfoto 2022-10-07 um 09 56 14" src="https://user-images.githubusercontent.com/109968361/194502700-c799a53b-649c-4470-9f55-08ad9bbe4810.png">

The Seconds are going 1 down, and then 1 up again. 
I was once to the pooint where all was working and made a commit, but when im Jumping back to that commit all Bugs still there

<img width="515" alt="Bildschirmfoto 2022-10-07 um 09 57 54" src="https://user-images.githubusercontent.com/109968361/194502841-57f5dd51-af1a-4b85-88c9-bfadc8110223.png">

my Branch that Im working on:
https://github.com/DennisWarchocki/capstone-project/tree/timer

I get displayed something like this:

<img width="1027" alt="Bildschirmfoto 2022-10-07 um 10 13 25" src="https://user-images.githubusercontent.com/109968361/194505948-c73e5d96-95b1-48fc-8cb7-82fc96a96522.png">

**Comment by mbosselmann** [comment, marked as answer]
Hello Dennis,
ok your bug report needed some time for me to figure out.😅 I'll list my answer:
 1. You wrote your url wrong. Currently you have to write `/Timer` to reach the route. If you rename the file in your `pages` folder by writing `Timer` in lower case, then you will reach the timer under the route `/timer` as you want.
2. You added `now` to your dependency array in your `useEffect`. That seems to be the reason why your timer always starts again. I think you added `now` to the dependency array, because there was a warning. If you put your const `now` and your `getTime` function inside the `useEffect` and add as dependencies `days`, `hours`, `minutes` and `days` it should work fine.
```
	useEffect(() => {
		const now = Date.now() + seconds + minutes + hours + days;

		function getTime() {
			const time = now - Date.now();
			if (time > 0) {
				setTimerDays(Math.floor(time / (24 * 60 * 60 * 1000)));

				setTimerHours(Math.floor((time / (1000 * 60 * 60)) % 24));

				setTimerMinutes(Math.floor((time / 1000 / 60) % 60));

				setTimerSeconds(Math.floor((time / 1000) % 60));
			} else {
				setTimerDays(0);
				setTimerHours(0);
				setTimerMinutes(0);
				setTimerSeconds(0);
			}
		}
		const interval = setInterval(() => getTime(now), 1000);
		return () => clearInterval(interval);
	}, [seconds, minutes, hours, days]);
```
I also think that you don't need the const `date` and `current`. You can hand over the parameter `now` to your getTime function.

3. The warning you got refers to the props of your `input` elements. As you used a regular html element and not a React component, only regular html attributes are allowed. Just change `userInputDays={userInputDays}` to `value={userInputDays}` (the same for hours, minutes and seconds) and the warning should not be displayed anymore.

**Reply by DennisWarchocki** [reply]
You are my savior Mareike, thank you so much. now everythings working out!
